test/utils/shoots/operation/operation.go

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package operation
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/ptr"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
+)
+
+// ReconciliationSuccessful checks if a shoot is successfully reconciled. In case it is not, it also returns a descriptive message stating the reason.
+func ReconciliationSuccessful(shoot *gardencorev1beta1.Shoot) (bool, string) {
```

Comment:
This function could also be in the `test/utils/shoots` package directly, if you want.

Reply:
I think the separation into separate packages is quite nice. Would your approach provide any additional benefits? :)

Reply:
Works for me, no strong opinion.